src/PhpSpec/Console/Application.php

```diff
@@ -166,18 +166,60 @@ protected function parseConfigurationFile(InputInterface $input)
             $paths = array($customPath);
         }
 
-        $config = array();
+        $config = $this->extractConfigFromFirstParsablePath($paths);
+
+        if ($homeFolder = getenv('HOME')) {
+            $config = array_replace_recursive($this->parseConfigFromExistingPath($homeFolder.'/.phpspec.yml'), $config);
+        }
+
+        return $config;
+    }
+
+    /**
+     * @param array $paths
+     *
+     * @return array
+     */
+    private function extractConfigFromFirstParsablePath(array $paths)
+    {
         foreach ($paths as $path) {
-            if ($path && file_exists($path) && $parsedConfig = Yaml::parse(file_get_contents($path))) {
-                $config = $parsedConfig;
-                break;
+
+            $config = $this->parseConfigFromExistingPath($path);
+
+            if (!empty($config)) {
+                return $this->addPathsToEachSuiteConfig(dirname($path), $config);
             }
         }
 
-        if ($homeFolder = getenv('HOME')) {
-            $localPath = $homeFolder.'/.phpspec.yml';
-            if (file_exists($localPath) && $parsedConfig = Yaml::parse(file_get_contents($localPath))) {
-                $config = array_replace_recursive($parsedConfig, $config);
+        return array();
+    }
+
+    /**
+     * @param string $path
+     *
+     * @return array
+     */
+    private function parseConfigFromExistingPath($path)
+    {
+        if (!file_exists($path)) {
+            return array();
+        }
+
+        return Yaml::parse(file_get_contents($path));
+    }
+
+    /**
+     * @param string $configDir
+     *
+     * @param array $config
+     *
+     * @return array
+     */
+    private function addPathsToEachSuiteConfig($configDir, $config)
+    {
+        if (isset($config['suites']) && is_array($config['suites'])) {
+            foreach ($config['suites'] as &$suiteConfig) {
+                $suiteConfig = str_replace('%paths.config%', $configDir, $suiteConfig);
```

Comment:
In my opinion it's better to explicitly set `$config['suites']` here rather than use the reference. It's fine in this case but can lead to unexpected side effects.